### PR TITLE
Upgrade runtime

### PR DESCRIPTION
**Pull request by Magnus-Kuhn**
# Readiness checklist

-   [ ] I added/updated tests.
-   [x] I ensured that the PR title is good enough for the changelog.
-   [x] I labeled the PR.

### CODE REVIEW

**Comment by jkoenig134** [comment]
@Milena-Czierlinski & @britsta could you please check if everything is covered?

File: test/spec.test.ts [hunk 1]
```diff
@@ -76,15 +76,16 @@ describe("test openapi spec against routes", () => {
...
                 .map((method) => method.toLocaleLowerCase())
                 .sort();
 
             expect(generatedMethods, `Path ${path} do not have the same methods`).toStrictEqual(manualMethods);
 
-            Object.keys(manualOpenApiSpec.paths[path]!).forEach((method) => {
+            Object.keys(manualOpenApiSpec.paths[path] ?? {}).forEach((method) => {
                 const key = method as "get" | "put" | "post" | "delete" | "options" | "head" | "patch";
-                const manualResponses = Object.keys(manualOpenApiSpec.paths[path]![key]?.responses ?? {});
+                // const manualResponses = Object.keys(manualOpenApiSpec.paths[path][key]?.responses ?? {});
```
Comment (britsta) [inline test/spec.test.ts:87]: Remove comment please.

File: packages/sdk/src/types/attributes/ConnectorAttribute.ts [hunk 2]
```diff
@@ -6,6 +6,7 @@ export interface ConnectorAttribute {
     succeededBy?: string;
     deletionInfo?: ConnectorAttributeDeletionInfo;
     shareInfo?: ConnectorAttributeShareInfo;
+    isDefault?: boolean;
```
Comment (britsta) [inline packages/sdk/src/types/attributes/ConnectorAttribute.ts:9]: I think this property should also be added to some Requests, for example, to the `GetOwnRepositoryAttributeRequest` or the `GetAttributesRequest`. In the `GetAttributesRequest`, there also seem to miss other properties.

File: package.json [hunk 3]
```diff
@@ -72,7 +72,7 @@
         "@js-soft/node-logger": "1.2.0",
         "@js-soft/ts-utils": "^2.3.3",
         "@nmshd/iql": "^1.0.2",
-        "@nmshd/runtime": "5.0.0-alpha.9",
+        "@nmshd/runtime": "5.0.0-alpha.14",
```
Comment (britsta) [inline package.json:75]: As I understand it, only the changes from release/v5 are included here, but not the changes from main. This means, for example, that `DeletionRequestSent` or `DeletionRequestRejected` have not yet been added to `ConnectorAttributeDeletionStatus`. We therefore need a second PR on main that incorporates the changes there.

Further things that I have noticed:
- Is there a reason, why the `deletionInfo` occurs in the `GetOwnRepositoryAttributesRequest`, but not in the `GetOwnSharedIdentityAttributesRequest` or `GetPeerSharedIdentityAttributesRequest`, for example? I would expect it to be the other way round. What can our Attribute deletion expert @Milena-Czierlinski say about this?
- Does anyone know why the `parentId` property does not occur in the `ConnectorAttribute`?

Comment (britsta) [inline package.json:75]: The PR https://github.com/nmshd/runtime/pull/217 does not seem to be taken into account in this PR.

**Review by britsta — COMMENTED** [review]
Questions for understanding:
- A route for changing the default RepositoryAttribute in the Connector should not be exposed, because it is a feature related to the App? So the `isDefault` property is set to `true` for the firstly created RepositoryAttribute of a given IdentityAttribute value type, but it should not be possible to change the default RepositoryAttribute for the time being with a Connector? @Milena-Czierlinski 
- Also, setting the communication language seems to be a feature more strongly related to the App, which is why there is no Connector route added for it, right? @Magnus-Kuhn

**Comment by Milena-Czierlinski** [comment]
> A route for changing the default RepositoryAttribute in the Connector should not be exposed, because it is a feature related to the App? So the isDefault property is set to true for the firstly created RepositoryAttribute of a given IdentityAttribute value type, but it should not be possible to change the default RepositoryAttribute for the time being with a Connector? @Milena-Czierlinski

Yes, this is how it is currently planned.

**Comment by Milena-Czierlinski** [comment]
> Is there a reason, why the deletionInfo occurs in the GetOwnRepositoryAttributesRequest, but not in the GetOwnSharedIdentityAttributesRequest or GetPeerSharedIdentityAttributesRequest, for example? I would expect it to be the other way round. What can our Attribute deletion expert @Milena-Czierlinski say about this?

Your expectation is correct. In the runtime it is implemented the other way around.

**Comment by Magnus-Kuhn** [comment]
> Also, setting the communication language seems to be a feature more strongly related to the App, which is why there is no Connector route added for it, right? 

Exactly.

File: packages/sdk/src/types/attributes/ConnectorAttribute.ts (hunk 2 above)
Comment (jkoenig134) [reply packages/sdk/src/types/attributes/ConnectorAttribute.ts]: IMO we should have a flag to stop the runtime automatically assigning defaults to attributes and then use this flag in the connector. In this case we can remove this everywhere in the connector.

Comment (jkoenig134) [reply packages/sdk/src/types/attributes/ConnectorAttribute.ts]: I close this as we don't want to expose more of the `isDefault` feature set to the connector.

File: package.json (hunk 3 above)
Comment (jkoenig134) [reply package.json]: @Magnus-Kuhn could you take care of these comments?

Comment (jkoenig134) [reply package.json]: tackeled in #236

File: packages/sdk/src/types/requests/ConnectorRequest.ts
Comment (jkoenig134) [inline packages/sdk/src/types/requests/ConnectorRequest.ts:1]: the naming in this file has been really confusing so I inlined some of the types to get rid of the naming issues

**Review by jkoenig134 — DISMISSED** [review]
LGTM. But wait for other reviews.

File: packages/sdk/src/types/requests/ConnectorRequest.ts [hunk 4]
```diff
@@ -7,8 +8,18 @@ export interface ConnectorRequest {
...
-    response?: ConnectorRequestResponse;
+    source?: {
+        type: "Message" | "RelationshipTemplate";
+        reference: string;
+    };
+    response?: {
+        createdAt: string;
+        content: ConnectorResponse;
+        source?: {
+            type: "Message" | "RelationshipChange";
```
Comment (Magnus-Kuhn) [inline packages/sdk/src/types/requests/ConnectorRequest.ts:19]:
```suggestion
            type: "Message" | "Relationship";
```

File: packages/sdk/src/types/requests/ConnectorResponse.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    "@type": "Response";
+    result: "Accepted" | "Rejected";
+    requestId: string;
+    items: (ConnectorResponseItemGroup | ConnectorResponseItem)[];
+}
+
+export interface ConnectorResponseItem {
+    "@type": string;
+    result: ConnectorResponseItemResult;
+    metadata?: object;
```
Comment (Magnus-Kuhn) [inline packages/sdk/src/types/requests/ConnectorResponse.ts:11]: Do we have metadata in responses? A quick runtime search says no

Comment (jkoenig134) [reply packages/sdk/src/types/requests/ConnectorResponse.ts]: nope, it's a relic

File: test/messages.test.ts [hunk 6]
```diff
@@ -114,7 +114,7 @@ describe("Message errors", () => {
                 "@type": "Mail",
                 subject: "A Subject",
                 body: "A Body"
-            }
+            } as any
```
Comment (Milena-Czierlinski) [inline test/messages.test.ts:117]: Can you create a `ConnectorMail` here instead?

File: test/attributes.test.ts [hunk 7]
```diff
@@ -133,7 +132,7 @@ describe("Attributes", () => {
         expect(notificationResponse.isSuccess).toBe(true);
 
         await syncUntilHasMessageWithNotification(client2, notificationResponse.result.notificationId);
-        await client2._eventBus!.waitForEvent<DataEvent<any>>("consumption.peerSharedAttributeSucceeded", (event: DataEvent<SuccessionEventData>) => {
+        await client2._eventBus!.waitForEvent<DataEvent<any>>("consumption.peerSharedAttributeSucceeded", (event: DataEvent<any>) => {
```
Comment (Milena-Czierlinski) [inline test/attributes.test.ts:135]: Why were the event types removed in this file?

File: packages/sdk/src/types/requests/ConnectorRequestContent.ts [hunk 8]
```diff
@@ -4,7 +4,7 @@ import { IdentityAttributeQuery } from "../attributes/IdentityAttributeQuery";
 import { RelationshipAttributeQuery } from "../attributes/RelationshipAttributeQuery";
 
 export interface ConnectorRequestContent {
-    "@type"?: string;
+    "@type"?: "Request";
```
[the remark this answers is not in the document]
Comment (Magnus-Kuhn) [reply packages/sdk/src/types/requests/ConnectorRequestContent.ts:7]: We use the ConnectorRequestContent also for creating a Request, where we don't burden the integrators with supplying this.

Also, I was terrified by your type annotation 😆

File: test/messages.test.ts (hunk 6 above)
Comment (jkoenig134) [reply test/messages.test.ts:117]: I want to turn of ts checks here, all good.

File: test/attributes.test.ts (hunk 7 above)
Comment (jkoenig134) [reply test/attributes.test.ts:135]: runtime was used in connector tests, major flaw.

File: packages/sdk/src/types/requests/ConnectorRequestContent.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (jkoenig134) [reply packages/sdk/src/types/requests/ConnectorRequestContent.ts:7]: Oh, I also was confused about this. I will update this to be more typesafe!

File: packages/sdk/src/types/relationshipTemplates/ConnectorRelationshipContentDerivation.ts
[the remark this answers is not in the document]
Comment (jkoenig134) [reply packages/sdk/src/types/relationshipTemplates/ConnectorRelationshipContentDerivation.ts:1]: true!

File: packages/sdk/src/types/requests/ConnectorRequestContent.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (Milena-Czierlinski) [reply packages/sdk/src/types/requests/ConnectorRequestContent.ts]: In this PR?

[the remark this answers is not in the document]
Comment (jkoenig134) [reply packages/sdk/src/types/requests/ConnectorRequestContent.ts]: Yep, I did.

**Review by Milena-Czierlinski — APPROVED** [review]

**Pull request merged**